src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/sdk/CloudSdk.java

```diff
@@ -43,20 +43,44 @@
   private Path sdkPath = null;
   private ProcessRunner processRunner = null;
 
-  public CloudSdk(File sdkPath) {
-    this(sdkPath, new DefaultProcessRunner());
+  /**
+   * Initializes an instance using the default location of Cloud SDK and default ProcessRunner.
+   */
+  public CloudSdk() {
+    this((File) null);
+  }
+
+  /**
+   * Initialize an instance using the default location of Cloud SDK and custom ProcessRunner.
+   */
+  public CloudSdk(ProcessRunner processRunner) {
+    this(null, processRunner);
+  }
+
+  /**
+   * Initialize an instance using the provided Cloud SDK path and default ProcessRunner.
+   */
+  public CloudSdk(File sdkPathFile) {
+    this(sdkPathFile, new DefaultProcessRunner());
   }
 
   /**
    * Initializes an instance using the specified SDK path and ProcessRunner.
    *
-   * @param sdkPath The home directory of Google Cloud SDK
+   * @param sdkPathFile The home directory of Google Cloud SDK. If null, will attempt to look for
+   *                    the SDK in known install locations.
    */
-  public CloudSdk(File sdkPath, ProcessRunner processRunner) {
-    if (sdkPath == null) {
-      throw new NullPointerException("sdkPath cannot be null - use PathResolver for defaults");
+  public CloudSdk(File sdkPathFile, ProcessRunner processRunner) {
+    if (sdkPathFile == null) {
+      Path discoveredSdkPath = PathResolver.INSTANCE.getCloudSdkPath();
+      if (discoveredSdkPath == null) {
+        throw new NullPointerException("Google Cloud SDK path was not provided and could not be"
```

Comment:
Is this really a NullPointerException? It seems like a more library specific exception. AppEngineException? or something else?